### DISCUSSION TITLE
Kiosk mode with multiple views jumping to first view

### DISCUSSION THREAD

**Discussion by fcassata** [NemesisRE/kiosk-mode#247] [Q&A] [answered]
I have different views and users for two NSPanel Pros that are running HA companion and the kiosk addon. All I do to enable kiosk mode to each view is adding "?kiosk" to the URL field in the view configuration.

This works fine as long as there is only one kiosk mode view, and that the view is the first one, neither of which is the case for me. All views with ?kiosk throw the user back to the first view.

Any ideas here?

**Comment by elchininet** [comment, marked as answer]
Hi @fcassata,
I think that what is happening is that you are adding the `?kiosk` parameter to the `URL` field of a view and pretending with this that when you click on a tab it will open that URL with the parameter at the end. Is that what you are trying?, because if it is that it will not work, but it will not work even if you don't have `kiosk-mode` installed and you try to add whatever parameter you want (like `?test` for example).

**Reply by fcassata** [reply]
That is indeed what I am doing and thought was the right way :)

Thanks for the pointer, I can see the exact same behaviour when I use a random parameter in the URL like you say.

What is the correct way to enable kiosk mode for a view that is opened via the companion apps?

**Reply by elchininet** [reply]
`kiosk-mode` is not enabled per view, it is enabled per dashboard. When you add `?kiosk` to a URL `kiosk-mode` is activated in that dashboard and it will remain activated if you navigate through all the views of that dashboard.

You can add this parameter to the URL that you attach to any button, sidebar item or link, But you cannot add it to the URL field of a view just because [it doesn't allow special characters](https://www.home-assistant.io/dashboards/views/#url-of-a-view).

**Reply by elchininet** [reply]
To notice this behaviour that I explained, just add `?hide_sidebar` to the URL in your browser ( In the companion App you cannot see the URL), after that navigate through your different views and you will notice how the URL changes to the correct views maintaining the parameters in the URL.

**Reply by fcassata** [reply]
Thanks for clarifying. So would it make more sense perhaps to create a seperate dashboard for each of these users? Can I enable kiosk mode through config without the url parameter?

**Reply by elchininet** [reply]
That would work in your specific case. You can use the [user_settings](https://github.com/NemesisRE/kiosk-mode?tab=readme-ov-file#user_settings) to set some options for cretain users.

**Reply by fcassata** [reply]
Thanks again for the explanations! I've changed the setup and all is working now, funny how I completely misunderstood this before!
